--- a/libqtile/backend/wayland/inputs.py	2023-09-26 20:19:11.710427183 +0900
+++ b/libqtile/backend/wayland/inputs.py	2023-09-26 20:19:29.988575454 +0900
@@ -142,12 +142,18 @@
         ("kb_variant", None, "Keyboard variant i.e. ``XKB_DEFAULT_VARIANT``"),
         ("kb_repeat_rate", 25, "Keyboard key repeats made per second"),
         ("kb_repeat_delay", 600, "Keyboard delay in milliseconds before repeating"),
+        ("send_events", 'enabled', "``'enabled'``, ``'disabled'`` or ``'disabled_on_external_mouse'``"),
     ]
 
     def __init__(self, **config: Any) -> None:
         configurable.Configurable.__init__(self, **config)
         self.add_defaults(InputConfig.defaults)
 
+    SENDEVENTS_MODE = {
+        "enabled": lib.LIBINPUT_CONFIG_SEND_EVENTS_ENABLED ,
+        "disabled": lib.LIBINPUT_CONFIG_SEND_EVENTS_DISABLED,
+        "disabled_on_external_mouse": lib.LIBINPUT_CONFIG_SEND_EVENTS_DISABLED_ON_EXTERNAL_MOUSE,
+    }
 
 class _Device(ABC, HasListeners):
     def __init__(self, core: Core, wlr_device: InputDevice):
@@ -364,3 +370,8 @@
                     lib.libinput_device_config_tap_set_button_map(
                         handle, TAP_MAPS.get(config.tap_button_map)
                     )
+
+        if config.send_events is not None:
+            lib.libinput_device_config_send_events_set_mode(
+                handle, SENDEVENTS_MODE.get(config.send_events)
+            ) 
