--- a/libqtile/backend/wayland/libinput_ffi_build.py.orig	2023-09-21 10:22:43.261473993 +0900
+++ b/libqtile/backend/wayland/libinput_ffi_build.py	2023-09-21 10:23:20.303392997 +0900
@@ -124,6 +124,16 @@
 enum libinput_config_status
 libinput_device_config_dwt_set_enabled(struct libinput_device *device,
                                        enum libinput_config_dwt_state enable);
+
+enum libinput_config_send_events_mode {
+    LIBINPUT_CONFIG_SEND_EVENTS_ENABLED,
+    LIBINPUT_CONFIG_SEND_EVENTS_DISABLED,
+    LIBINPUT_CONFIG_SEND_EVENTS_DISABLED_ON_EXTERNAL_MOUSE,
+};
+
+enum libinput_config_status
+libinput_device_config_send_events_set_mode(struct libinput_device *device,
+                                            uint32_t mode);
 """
 
 libinput_ffi = FFI()
